copy_req_files_to_vm.sh

```diff
@@ -0,0 +1,65 @@
+#! /usr/bin/env bash
+
+# Script name: copy_proj_to_vm.sh
+# Description: Run this script BEFORE your run: setup_mysql.sh & setup_machine.sh
+# This script transfers those 2 setup files & the project files &
+# folders for the create-jazz-lyric Python Flask web app to the deployment machine.
+# Author: Kim Lew
+
+# --- Using Vagrant Configuration File ----------------------------------------
+# Assumes: vagrant-ssh-config was created as early step in README.
+
+VAG_SSH_CFG='vagrant-ssh-config'
+SRC_REQ_FILES_DIR='/Users/kimlew/code/vagrant_vm_centos7_create_jazz/'  # On Mac.
+SRC_PROJ_DIR='/Users/kimlew/code/vagrant_vm_centos7_create_jazz/pythonapp-createjazzlyric/'  # On Mac.
+DEST_REQ_FILES_AT_ROOT_DIR='/home/vagrant/'  # VM root directory.
+DEST_PROJ_DIR='/home/vagrant/pythonapp-createjazzlyric/'  # On VM.
+
+# --- Files needed at root directory on VM to set up VM. ---
+scp -F "${VAG_SSH_CFG}" "${SRC_REQ_FILES_DIR}"setup_mysql.sh vagrant@default:"${DEST_ROOT_DIR}"setup_mysql.sh
+scp -F "${VAG_SSH_CFG}" "${SRC_REQ_FILES_DIR}"setup_machine.sh vagrant@default:"${DEST_ROOT_DIR}"setup_machine.sh
```

Comment:
I noticed several days ago when you were showing me this script that you included a trailing slash in the value of `SRC_REQ_FILES_DIR`. Did you do that just so you wouldn't have to insert a slash after the variable interpolation like this:

```
"${SRC_REQ_FILES_DIR}"/setup_mysql.sh
```

I ask because I usual dislike having to remember to put a trailing slash on the value of the variable, because it's not obvious from the variable name and it's easy to forget. But you might have a good reason for doing it that I'm not seeing.

Reply:
I DID add the trailing / to the ends of the paths for the directory variables because I always forget to manually add the trailing / in the actual command.
I also find it harder to read when it is all together in the command. For example:
Easier to Read:  `scp -F "${VAG_SSH_CFG}" "${SRC_REQ_FILES_DIR}"setup_mysql.sh vagrant@default:"${DEST_ROOT_DIR}"setup_mysql.sh`
Harder to Read: `scp -F "${VAG_SSH_CFG}"**/**"${SRC_REQ_FILES_DIR}"setup_mysql.sh vagrant@default:"${DEST_ROOT_DIR}"setup_mysql.sh`

Reply:
I think you can leave it as is as long as you are consistent about `_DIR` variables having values that end in slashes.

You're right that there are mixed opinions on this, and partly it's because paths are a little quirky. Ordinarily the slash `/` is just a separator, except when it appears all by itself and means the root directory. And, as you have learned, different tools (like `rsync`) ascribe special meaning to a trailing slash, but most tools ascribe no special meaning to a trailing slash.

You said

> Harder to Read:
> scp -F "${VAG_SSH_CFG}"/"${SRC_REQ_FILES_DIR}"setup_mysql.sh vagrant@default:"${DEST_ROOT_DIR}"setup_mysql.sh
> 
> Easier to Read:
> scp -F "${VAG_SSH_CFG}" "${SRC_REQ_FILES_DIR}"setup_mysql.sh vagrant@default:"${DEST_ROOT_DIR}"setup_mysql.sh
> 

But did you mean for the "Harder to Read" example to be:

```
scp -F "${VAG_SSH_CFG}" "${SRC_REQ_FILES_DIR}"/setup_mysql.sh vagrant@default:"${DEST_ROOT_DIR}"/setup_mysql.sh
```

I ask because I was suggesting putting a slash `/` after the variable substitutions for `${SRC_REQ_FILES_DIR}` and `${DEST_ROOT_DIR}`, like above, and leaving it out of the values of the `_DIR` variables.

The reason I suggested this is I would personally find it hard to remember that if I change the values assigned to `SRC_REF_FILES_DIR` and `DEST_ROOT_DIR` that I have to add trailing slashes. The names of the variables don't suggest to me that they need a trailing slash, so I might forget. However, you personally can adopt the convention that a `..._DIR` variable always has a value that ends in a slash if you like. If you adopt that convention and use it consistently, then that's fine too.

One commenter pointed out that if you keep things the way you have them then you could use an empty value for `SRC_REF_FILES_DIR` to mean the present working directory. Personally I think that's a little fragile and it "works by accident". You can refer to the present working directory more directly with `${PWD}` and I would prefer that, but that's just me.